File: frontend/src/api/message.ts
import request from './index'

export interface Message {
  id: number
  user_id: number
  title: string
  content?: string
  type: string
  is_read: boolean
  related_task_id?: number
  created_at: string
}

export interface MessageListResponse {
  total: number
  items: Message[]
}

export interface MessageUnreadCountResponse {
  unread_count: number
}

/**
 * 获取消息列表
 */
export function getMessages(params?: {
  type?: string
  is_read?: boolean
  page?: number
  page_size?: number
}): Promise<MessageListResponse> {
  return request.get('/api/v1/messages', { params })
}

/**
 * 获取未读消息数量
 */
export function getUnreadCount(): Promise<MessageUnreadCountResponse> {
  return request.get('/api/v1/messages/unread-count')
}

/**
 * 标记消息为已读
 */
export function markAsRead(messageId: number): Promise<Message> {
  return request.post(`/api/v1/messages/${messageId}/read`)
}

/**
 * 标记所有消息为已读
 */
export function markAllAsRead(type?: string): Promise<{ message: string }> {
  return request.post('/api/v1/messages/read-all', null, { params: type ? { type } : {} })
}

/**
 * 删除消息
 */
export function deleteMessage(messageId: number): Promise<void> {
  return request.delete(`/api/v1/messages/${messageId}`)
}

File: backend/app/schemas/message.py
"""消息通知相关模式"""
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class MessageCreate(BaseModel):
    """创建消息请求"""
    user_id: int = Field(..., description="接收用户ID")
    title: str = Field(..., description="消息标题")
    content: Optional[str] = Field(None, description="消息内容")
    type: str = Field(..., description="消息类型")
    related_task_id: Optional[int] = Field(None, description="关联任务ID")


class MessageResponse(BaseModel):
    """消息响应"""
    id: int
    user_id: int
    title: str
    content: Optional[str]
    type: str
    is_read: bool
    related_task_id: Optional[int]
    created_at: datetime

    class Config:
        from_attributes = True


class MessageListResponse(BaseModel):
    """消息列表响应"""
    total: int
    items: list[MessageResponse]


class MessageUnreadCountResponse(BaseModel):
    """未读消息数量响应"""
    unread_count: int
